scripts/bin/update-packages

```diff
@@ -292,6 +302,38 @@ _run_update() {
 	fi
 }
 
+_create_pending_prs_for_auto_update_groups() {
+	local changed_files current_ref versions title
+	current_ref="$(git rev-parse --abbrev-ref HEAD)"
+	for group in ${AUTO_UPDATE_GROUPS[@]}; do
+		# switch to potential PR branch if it exists.
+		git checkout "auto-update/${group}/${GITHUB_RUN_ID}" || continue
+
+		# obtain all build.sh files compared to origin.
+		readarray -t changed_files <<< "$(git diff --name-only $(git merge-base "${current_ref}" HEAD) HEAD | grep '/build.sh$')"
+
+		# read all versions
+		readarray -t versions < <(
+			for _build_sh in "${changed_files[@]}"; do
+			( # do everything in subshell to avoid variable mess
```

Comment:
Isn't the surrounding `<()` already a subshell?

Reply:
Yes, but we do not want any variable from these multiple `build.sh` files to be reused and cause any side effect in other `build.sh` files.